src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/ManagedTypeInfo.cs

```diff
@@ -18,6 +15,19 @@ public abstract record ManagedTypeInfo(string FullTypeName, string DiagnosticFor
         private TypeSyntax? _syntax;
         public TypeSyntax Syntax => _syntax ??= SyntaxFactory.ParseTypeName(FullTypeName);
 
+        public virtual bool Equals(ManagedTypeInfo other)
+        {
+            return other is not null
+                && Syntax.IsEquivalentTo(other.Syntax)
+                && FullTypeName == other.FullTypeName
+                && DiagnosticFormattedName == other.DiagnosticFormattedName;
+        }
+
+        public override int GetHashCode()
+        {
+            return Syntax.GetHashCode() ^ FullTypeName.GetHashCode() ^ DiagnosticFormattedName.GetHashCode();
```

Comment:
I think that the `Syntax.GetHashCode()` call here should be dropped. It doesn't look like `SyntaxNode` overrides `GetHashCode()`, so we'll end up with mismatching hash codes for equal values, which breaks a lot of things.

```suggestion
            return FullTypeName.GetHashCode() ^ DiagnosticFormattedName.GetHashCode();
```